src/sinks/pulsar.rs

```diff
@@ -59,6 +58,9 @@ pub struct PulsarSinkConfig {
         skip_serializing_if = "crate::serde::skip_serializing_if_default"
     )]
     pub acknowledgements: AcknowledgementsConfig,
+
+    /// Log field to use as Pulsar message key
+    key_field: Option<String>,
```

Comment:
maybe this should be named `partition_key_field` since there is also an "ordering key" which could be confused with this?

Reply:
From my understanding, there is only 1 kind of key on a message. This key is used for ordering and consumption.

```rust
// pulsar-rs/producer.rs
    pub fn with_partition_key<S: Into<String>>(mut self, partition_key: S) -> Self {
        self.partition_key = Some(partition_key.into());
        self
    }
    /// sets the message's ordering key for key_shared subscription
    pub fn with_ordering_key<S: Into<String>>(mut self, partition_key: S) -> Self {
        self.partition_key = Some(partition_key.into());
        self
    }

    /// sets the message's partition key
    ///
    /// this is the same as `with_partition_key`, this method is added for
    /// more consistency with other clients
    pub fn with_key<S: Into<String>>(mut self, partition_key: S) -> Self {
        self.partition_key = Some(partition_key.into());
        self
    }
 ```
 But I can rename it `partition_key_field`

Reply:
That's interesting. I was just looking at the crate documentation, so I didn't realize it's setting the same value. In that case, I'm fine with the current naming.